Makefile

```diff
@@ -14,29 +14,22 @@ type: pytype mypy
 
 lint:
 	# stop the build if there are Python syntax errors or undefined names
-	# see https://lintlyci.github.io/Flake8Rules/
-	flake8 ${LINT_PATHS} --count --select=E9,F63,F7,F82 --show-source --statistics
-	# exit-zero treats all errors as warnings.
-	flake8 ${LINT_PATHS} --count --exit-zero --statistics
-
-ruff:
-	# stop the build if there are Python syntax errors or undefined names
-	# see https://lintlyci.github.io/Flake8Rules/
+	# see https://www.flake8rules.com/
 	ruff ${LINT_PATHS} --select=E9,F63,F7,F82 --show-source
```

Comment:
`--select` seems to redundant with the config in `pyproject.toml`

Reply:
Also, I think it is `ruff check`, not just `ruff`. (It's weird that it doesn't raise an error).

Reply:
Why do you need to run 2 checks instead of one? Ie, why not just `--show-source --exit-zero`

Reply:
> eems to redundant with the config in pyproject.toml
> Why do you need to run 2 checks instead of one? Ie, why not just --show-source --exit-zero

The first command allow to check errors that would crash the code (undefined variables, syntax error, missing imports, ...).
The second one (define in the config) is about warnings and nice to have, the code will run.
So it's ok that the second check fail (hence `--exit-zero`) but not the first one.

> Also, I think it is ruff check, not just ruff. (It's weird that it doesn't raise an error).

`check` is the default command so `ruff` and `ruff check` are equivalent

Reply:
btw, as ruff is really fast, I should probably put the linter before the type checking, no?

Reply:
> btw, as ruff is really fast, I should probably put the linter before the type checking, no?

agree

Reply:
> `--select` seems to redundant with the config in `pyproject.toml`

@araffin can you just clarify this?

Reply:
If I don't use select, it's gonna run all the rules defines in the `pyproject.toml` and fail at the first warning.